shellclear/src/clearer.rs

```diff
@@ -0,0 +1,156 @@
+use std::collections::HashMap;
+use std::fmt::Write;
+use std::fs::write;
+use std::time::Instant;
+
+use anyhow::Result;
+
+use crate::data::FindingSensitiveCommands;
+use crate::shell::Shell;
+use crate::ShellContext;
+
+#[derive(Default)]
+pub struct Clearer {}
+
+impl Clearer {
+    pub fn write_findings(
+        self,
+        shells_context: &[ShellContext],
+        findings: &[FindingSensitiveCommands],
+        remove: bool,
+    ) -> Result<()> {
+        let findings_per_shell = self.group_findings_by_shell(findings);
+
+        for context in shells_context {
+            let start = Instant::now();
+            let mut cleared_history: String = String::new();
+
+            for &r in findings_per_shell
+                .get(&context.history.shell)
+                .unwrap_or(&vec![])
+            {
+                if remove {
+                    if r.sensitive_findings.is_empty() {
+                        writeln!(&mut cleared_history, "{}", r.data)?;
+                    }
+                } else {
+                    writeln!(&mut cleared_history, "{}", r.data)?;
+                }
+            }
+
+            if !cleared_history.is_empty() {
+                write(&context.history.path, cleared_history)?;
+                log::debug!(
+                    "time elapsed for backup existing file and write a new history to shell : {:?}",
+                    start.elapsed()
+                );
+            }
+        }
+
+        Ok(())
+    }
+
+    fn group_findings_by_shell(
```

Comment:
Can we change the engine result to return the finding result like
```rs
pub struct Findings {
    engine_kind: HashMap<Shell, Vec<&FindingSensitiveCommands>>
}
```
this will more generic, and keep the logic inside the engine any be more extendable